src/python_framework_v02/troute/network/musking/mc_reach_structs.c

```diff
@@ -0,0 +1,51 @@
+#include <stdlib.h>
+#include "../reach_structs.h"
+
+void init_mc_reach(_Reach* reach, int num_segments)
+{
+  reach->reach.mc_reach.num_segments = num_segments;
+  reach->reach.mc_reach._segments = (_MC_Segment*) malloc(sizeof(_MC_Segment)*(num_segments));
+}
+
+void free_mc_reach(_Reach* reach)
+{
+  if( reach != NULL && reach->reach.mc_reach._segments != NULL)
+    free(reach->reach.mc_reach._segments);
+}
+
+void set_mc_segment(_Reach* reach, int index, long id,
+    float dt, float dx, float bw, float tw, float twcc,
+    float n, float ncc, float cs, float s0,
+    float qdp, float velp, float depthp)
+{
+  if(index > -1 && index < reach->reach.mc_reach.num_segments)
+  {
+    _MC_Segment segment;
+    //TODO what about segment id???
+    segment.id = id;
+    segment.dt = dt;
+    segment.dx = dx;
+    segment.bw = bw;
+    segment.tw = tw;
+    segment.twcc = twcc;
+    segment.n = n;
+    segment.ncc = ncc;
+    segment.cs = cs;
+    segment.s0 = s0;
+    segment.qdp = qdp;
+    segment.velp = velp;
+    segment.depthp = depthp;
+    reach->reach.mc_reach._segments[index] = segment;
+  }
+  //FIXME else what?
+}
+
+_MC_Segment get_mc_segment(_Reach* reach, int index)
+{
+  _MC_Segment seg;
+  if( reach != NULL && index > -1 && index < reach->reach.mc_reach.num_segments){
```

Comment:
Should carefully think about what is returned when the reach/seg is NOT valid.  A  default segment is created, its values are not guaranteed to be in any meaninful state, and its possible that it gets returned...